packages/core/test/unit/test-runner/MaxTestRunnerReuseDecorator.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { TestRunner2 } from '@stryker-mutator/api/test_runner';
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { factory } from '@stryker-mutator/test-helpers';
+
+import TestRunnerDecorator from '../../../src/test-runner/TestRunnerDecorator';
+import MaxTestRunnerReuseDecorator from '../../../src/test-runner/MaxTestRunnerReuseDecorator';
+
+describe(MaxTestRunnerReuseDecorator.name, () => {
+  let testRunner: sinon.SinonStubbedInstance<Required<TestRunner2>>;
+  const runOptions = factory.mutantRunOptions({ timeout: 23 });
+
+  beforeEach(() => {
+    testRunner = factory.testRunner();
+  });
+
+  const getSut = (maxTestRunnerReuse: number) => {
+    const sut = new MaxTestRunnerReuseDecorator(() => testRunner, { maxTestRunnerReuse: maxTestRunnerReuse });
+    sinon.spy(sut, 'dispose');
+    return sut;
+  };
+
+  it('should not override `init`', () => {
+    expect(getSut(1).init).to.be.eq(TestRunnerDecorator.prototype.init);
+  });
+
+  it('should override `dispose`', () => {
+    expect(getSut(1).dispose).to.not.be.eq(TestRunnerDecorator.prototype.dispose);
+  });
+
+  it('should not override `dryRun`', () => {
+    expect(getSut(1).dryRun).to.be.eq(TestRunnerDecorator.prototype.dryRun);
+  });
+
+  it('should pass through resolved values', async () => {
+    const sut = getSut(0);
+    const options = factory.mutantRunOptions({ timeout: 23 });
+    const expectedResult = factory.completeDryRunResult();
+    testRunner.mutantRun.resolves(expectedResult);
+    const result = await sut.mutantRun(options);
+    expect(testRunner.mutantRun).to.have.been.calledWith(options);
+    expect(result).to.eq(expectedResult);
+  });
+
+  it('should not dispose worker if restartAfterRuns is set to 0', async () => {
+    const sut = getSut(0);
+    const expectedResult = factory.completeDryRunResult();
+    testRunner.mutantRun.resolves(expectedResult);
+
+    await sut.mutantRun(runOptions);
+    const result = await sut.mutantRun(runOptions);
+
+    expect(sut.dispose).to.have.been.callCount(0);
+    expect(result).to.eq(expectedResult);
+  });
+
+  it('should dispose worker on second run if restartAfterRuns is set to 1', async () => {
```

Comment:
I think we're still missing two tests here.

1. A second retry. That one tests that `this.runs` is correctly reset after `dispose`.
1. Run once, dispose, and then see that `this.runs` is correctly reset.